### PR TITLE
29173 dual record display

### PR DESCRIPTION
**Pull request by AlejandroColinaGSA**
## Description
Provides the HTML required to display a newer record next to its older counterpart in a search results list. It highlights the differences and has an information dropdown that explains what this item is.

## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [ x] Internet Explorer 11
- [ x] Edge
- [ x] Chrome
- [ x] Firefox
- [ x] Safari

## Checklist:

- [ x] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ x] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ x] My code is 508 compliant as tested by AMP and JAWS
- [ x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.html [hunk 1]
```diff
@@ -1,15 +1,43 @@
-<div class="sds-result-item">
+<hr *ngIf="!model.isDualRecord" class="thin" />
```
Comment (davereed) [inline libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.html]: ngif is not needed, the 'dual display' should be a single record

File: libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.html [hunk 2]
```diff
@@ -1,15 +1,43 @@
...
+      </a>
+      <span *ngIf="model.isDualRecord" class="height-4">
+        <span class="float-left margin-right-1">
+          <fa-icon
+          [fixedWidth]="true"
+          [icon]="['sds', 'new-version']"
+          transform="grow-10"></fa-icon>
+        </span>
+        <span class="dual-record--title">
+          Dual Record</span>
```
Comment (davereed) [inline libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.html]: Remove junk/test output

File: libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.html [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
+          <div class="sds-popup__content">
+            <fa-icon
+                [fixedWidth]="true"
+                [icon]="['sds', 'information-circle']"
+                transform="grow-5"
+                class="float-right"></fa-icon>
+            <ul>
+              <li>Lorem ipsum</li>
+              <li>Lorem ipsum</li>
+            </ul>
```
Comment (davereed) [inline libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.html]: Remove junk/test output

File: libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample.component.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <sds-page class="condensed">
-  <sds-search-result-list [model]="items">
+  <sds-search-result-list [model]="items" [divider]=none>
```
Comment (davereed) [inline libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample.component.html]: divider is boolean

File: libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample.component.ts [hunk 5]
```diff
@@ -2,21 +2,20 @@ import { Component, OnInit, ViewChild, AfterViewInit, ChangeDetectorRef } from '
...
-
-  items= [
+  items = [
     { text: 'First', id: 1 },
     { text: 'Second', id: 2 },
     { text: 'Third', id: 3 },
     { text: 'Fourth', id: 4 },
-    { text: 'Fifth', id: 5 }
+    { text: 'Fifth', id: 5 },
+    { text: 'Sixth', id: 6, isDualRecord: true },
```
Comment (davereed) [inline libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample.component.ts]: Do not add additional records, the dual display should simply add on data to existing items.

File: libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.ts [hunk 6]
```diff
@@ -10,6 +10,7 @@ export class SearchResultListSampleDataComponent {
   constructor(private change: ChangeDetectorRef) { }
 
   @Input() model;
+  @Input() dualDisplay;
```
Comment (davereed) [inline libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.ts]: do we need this input? I think model information is enough.

Comment (AlejandroColinaGSA) [reply libs/component-sample/src/lib/feature/search-result-list-sample/search-result-list-sample-data.component.ts]: That is from an earlier commit, but it is not the current state of the branch:

https://github.com/GSA/sam-design-system/blob/29173-dual-record-display/libs/layout-sample/src/lib/ui/search-result-list-sample-data/search-result-list-sample-data.component.ts

**Pull request closed without merging**